addons/account/populate/account_move.py

```diff
@@ -0,0 +1,214 @@
+            """Get a random partner depending on the company and the move_type.
+
+            The first 3/5 of the available partners are used as customer
+            The last 3/5 of the available partners are used as suppliers
+            It means 1/5 is both customer/supplier
+            -> Same proportions as in account.payment
+            :param random: seeded random number generator.
+            :param values (dict): the values already selected for the record.
+            :return (int, bool): the id of the partner randomly selected if it is an invoice document
+                                 False if it is a Journal Entry.
+            """
+            move_type = values['move_type']
+            company_id = values['company_id']
+            partner_ids = search_partner_ids(company_id)
+
+            if move_type in self.get_sale_types(include_receipts=True):
+                return random.choice(partner_ids[:math.ceil(len(partner_ids)/5*2)])
+            if move_type in self.get_purchase_types(include_receipts=True):
+                return random.choice(partner_ids[math.floor(len(partner_ids)/5*2):])
+            return False
+
+        company_ids = self.env['res.company'].search([('chart_template_id', '!=', False), ('id', 'in', self.env.registry.populated_models['res.company'])])
+
+        return [
+            ('move_type', populate.randomize(['entry', 'in_invoice', 'out_invoice', 'in_refund', 'out_refund', 'in_receipt', 'out_receipt'], [0.2, 0.3, 0.3, 0.07, 0.07, 0.03, 0.03])),
+            ('company_id', populate.randomize(company_ids.ids)),
+            ('currency_id', populate.randomize(self.env['res.currency'].search([('active', '=', True)]).ids)),
+            ('journal_id', populate.compute(get_journal)),
+            ('date', populate.randdatetime(relative_before=relativedelta(years=-4), relative_after=relativedelta(years=1))),
+            ('invoice_date', populate.compute(get_invoice_date)),
+            ('partner_id', populate.compute(get_partner)),
+            ('line_ids', populate.compute(get_lines)),
+        ]
+
+    def _populate(self, size):
+        records = super()._populate(size)
+        _logger.info('Posting Journal Entries')
+        to_post = records.filtered(lambda r: r.date < fields.Date.today())
+        to_post.post()
+
+        _logger.info('Registering Payments for Invoices and Bills')
+        random = populate.Random('account.move+register_payment')
+        for invoice in to_post:
+            if invoice.is_invoice() and random.uniform(0, 1) < 0.9:  # 90% of invoices are at least partialy paid
+                payment_wizard = self.env['account.payment.register'].with_context(active_model='account.move', active_ids=invoice.ids).create({})
+                if random.uniform(0, 1) > 0.9:  # 90% of paid invoices have the exact amount, others vary a little
+                    payment_wizard.amount *= random.uniform(0.5, 1.5)
+                payment_wizard._create_payments()
```

Comment:
this part is costly, and does not cover all use cases.

Maybe we could do:
* 10% of invoices fully paid via register payment wizard
* 10% of invoices partially paid via register payment wizard
* most of the reconciliation created by directly populating account.partial.reconcile with a few account.partial.reconcile created to cover more complex use case (a payment matched with several invoices, an invoice matched with several payments, several payments matching several invoices)

Or simply populate the account.partial.reconcile and keep the register payment for the fully paid invoices